test/unittest/integration_tests/integration_test_fsu.cpp

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Donghak Park <donghak.park@samsung.com>
+ *
+ * @file   integration_test_fsu.cpp
+ * @date   20 Dec 2024
+ * @brief  Unit Test for Asynch FSU
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Donghak Park <donghak.park@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+
+#include <app_context.h>
+#include <array>
+#include <chrono>
+#include <ctime>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <layer.h>
+#include <memory>
+#include <model.h>
+#include <optimizer.h>
+#include <sstream>
+#include <vector>
+
+using LayerHandle = std::shared_ptr<ml::train::Layer>;
+using ModelHandle = std::unique_ptr<ml::train::Model>;
+
+template <typename T>
+static std::string withKey(const std::string &key, const T &value) {
+  std::stringstream ss;
+  ss << key << "=" << value;
+  return ss.str();
+}
+
+template <typename T>
+static std::string withKey(const std::string &key,
+                           std::initializer_list<T> value) {
+  if (std::empty(value)) {
+    throw std::invalid_argument("empty data cannot be converted");
+  }
+
+  std::stringstream ss;
+  ss << key << "=";
+
+  auto iter = value.begin();
+  for (; iter != value.end() - 1; ++iter) {
+    ss << *iter << ',';
+  }
+  ss << *iter;
+
+  return ss.str();
+}
+
+TEST(fsu, simple_fc) {
+
+  std::unique_ptr<ml::train::Model> model = ml::train::createModel(
+    ml::train::ModelType::NEURAL_NET, {withKey("loss", "mse")});
+
+  model->addLayer(ml::train::createLayer(
+    "input", {withKey("name", "input0"), withKey("input_shape", "1:1:320")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
```

Comment:
What about updating this duplicate code blocks with loop?

Reply:
Good idea i will fix that !